imports/math/shared.lua

```diff
@@ -134,4 +134,46 @@ function math.clamp(val, lower, upper) -- credit https://love2d.org/forums/viewt
     return math.max(lower, math.min(upper, val))
 end
 
+---Linearly interpolates between two values over a specified duration.
+---@param start number|table|vector2|vector3|vector4  -- The starting value of the interpolation.
+---@param finish number|table|vector2|vector3|vector4 -- The ending value of the interpolation.
+---@param duration number                             -- The duration over which to interpolate.
+---@return function                                   -- a function that returns interpolated values over time.
+function math.lerp(start, finish, duration)
+	local startTime = GetGameTimer()
+	local typeStart = type(start)
+
+	if typeStart ~= type(finish) then
+		lib.print.error("start and finish must be of the same type")
```

Comment:
Should probably be more descriptive, and use the `error` function for full stack-trace and error handling. See some other error messages I've written for inspiration.